packages/upload-api/src/usage/record.js

```diff
@@ -0,0 +1,40 @@
+import * as API from '../types.js'
+import * as Provider from '@ucanto/server'
+import { Usage } from '@web3-storage/capabilities'
+
+/** @param {API.UsageServiceContext} context */
+export const provide = (context) =>
+  Provider.provide(Usage.record, (input) => record(input, context))
+
+/**
+ * @param {API.Input<Usage.record>} input
+ * @param {API.UsageServiceContext} context
+ * @returns {Promise<API.Result<API.EgressRecordSuccess, API.EgressRecordFailure>>}
+ */
+const record = async ({ capability, invocation }, context) => {
+  const provider = /** @type {`did:web:${string}`} */ (
+    invocation.audience.did()
+  )
+  const consumerResponse = await context.provisionsStorage.getConsumer(
+    provider,
+    capability.with
+  )
+  if (consumerResponse.error) {
+    return {
```

Comment:
This actually returns `ConsumerNotFound` error when there's no subscription setup. It's arguably more useful than "Failed to get consumer".

```suggestion
    return consumerResponse
```

Really we need to be better at defining error types, as the failure type for `getConsumer` is currently the catchall `Failure`.

If you're up for it, it would be good to add a type for `ConsumerNotFound` and include it as a possible value for `EgressRecordFailure`.

i.e.
```ts
type EgressRecordFailure = ConsumerNotFound | Ucanto.Failure
```

Reply:
That makes sense to me. I will update the types. Thanks